core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -571,6 +572,34 @@ class FsHistoryProviderSuite extends SparkFunSuite with BeforeAndAfter with Matc
     }
  }
 
+  test("log without read permission should be filtered out before actual reading") {
+    class TestFsHistoryProvider extends FsHistoryProvider(createTestConf()) {
```

Comment:
I think you'll need this from another test in this same file:

```
    // setReadable(...) does not work on Windows. Please refer JDK-6728842.
    assume(!Utils.isWindows)
```

In fact shouldn't this test be merged with the test for SPARK-3697?